src/utils/layout-helpers.ts

```diff
@@ -196,6 +201,16 @@ export function cached(_proto: unknown, prop: string, desc: PropertyDescriptor)
 
 async function getRecursiveInRepoAddonRoots(root: string, roots: string[]) {
   const packageData = await asyncGetPackageJSON(root);
+
+  // names are required for packages
+  if (!packageData.name) return [];
+
+  if (SEEN.has(packageData.name)) {
```

Comment:
we should have independent `SEEN` on `Project` instance, shared will break "second" initialized project deps

Reply:
done!